components/renku_data_services/utils/core.py

```diff
@@ -12,3 +15,22 @@ def get_ssl_context():
     if custom_cert_file:
         context.load_verify_locations(cafile=custom_cert_file)
     return context
+
+
+def merge_api_specs(*args):
+    """Merges API spec files into a single one."""
+    merger = Merger(
+        type_strategies=[(list, "append_unique"), (dict, "merge"), (set, "union")],
+        fallback_strategies=["override"],
+        type_conflict_strategies=["override_if_not_empty"],
+    )
+
+    merged_spec: dict[str, Any]
+    merged_spec = functools.reduce(merger.merge, args, dict())
+
+    # Remove duplicate entries in `.servers`
+    server_urls: Set[str]
+    server_urls = set((obj["url"] for obj in merged_spec.get("servers", [])))
+    merged_spec["servers"] = [dict(url=url) for url in server_urls]
```

Comment:
I don't think you need this - check https://renku-ci-ds-77.dev.renku.ch/swagger/?urls.primaryName=data%20service

![image](https://github.com/SwissDataScienceCenter/renku-data-services/assets/3353586/b469b00a-4995-4bef-abc7-1e457ac2d291)

Reply:
Done in 2759e23cf9c07ba1bd20ce14df1317bc1fe76934.